### ISSUE TITLE
Error prone implementation of test_get_config_time

### ISSUE THREAD

**Issue by q-nk** [Nitrokey/nethsm-sdk-py#36]
The test `test_get_config_time`, which is implemented in `get_config_time` is error prone.
The problem is the following code 

```
def get_config_time(nethsm):
    time_nethsm_str = nethsm.get_config_time()
    # parse time_nethsm_str to datetime.datetime
    # 2023-09-22T14:46:12Z
    time_nethsm = datetime.datetime.strptime(time_nethsm_str, "%Y-%m-%dT%H:%M:%SZ")

    time_now = datetime.datetime.now(datetime.timezone.utc)

    assert datetime.datetime(
        time_nethsm.year,
        time_nethsm.month,
        time_nethsm.day,
        time_nethsm.hour,
        time_nethsm.minute,
    ) == datetime.datetime(
        time_now.year, time_now.month, time_now.day, time_now.hour, time_now.minute
    )
```

from https://github.com/Nitrokey/nethsm-sdk-py/blob/main/tests/test_nethsm_config.py#L33.

If the execution time between 

```
time_nethsm_str = nethsm.get_config_time()
```

and

```
time_now = datetime.datetime.now(datetime.timezone.utc)
```

is big enough, that the second time sample derives a different time, the test fails.

Solving this issue is tricky, as it depends on latency and execution time of the code.

To mitigate the problem, the comparison could be made with respect to an delta.
But this is no clean solution and may even makes the test useless, as the NetHSMs time can diverge within these delta.

**Comment by jans23** [comment]
It's still the best we can do. If possible, handle the time in Unix time format so that we can simply define the delta instead of dealing with the complexity of being at the edge between two days or two months.

**Comment by ansiwen** [comment]
It's already not comparing the seconds, but of course the minutes can flip just in that moment.

Let's take a delta and it must be less than a second. If it is more, we have a problem in any case.

**Comment by q-nk** [comment]
You're right, I overlooked that.

As discussed, I am going to implement a delta based comparison.
But I in my opinion the delta check must be less than 2.0 seconds, because the timestamp from the NetHSM is with a second granularity while the Python timestamp is more accurate. This can lead to a delta > 1.0.

Example: 
At real time 0.99 the NetHSM answers the time request with timestamp 0.
At real time 1.01 the time measured in the Python application is 1.01.
Therefore the absolute difference is > 1.0.

Conducting the measurements a few times, I saw differences up to 1.55.
So I think checking if the delta is < 2.0 is reasonable. 

@ansiwen Are you ok with a delta of up to 2.0 seconds?
